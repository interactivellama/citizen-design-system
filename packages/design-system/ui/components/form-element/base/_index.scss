// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/**
 * @summary Creates layout for a form element
 *
 * @name base
 * @selector .slds-form-element
 * @restrict div, fieldset, li
 * @variant
 */
.slds-form-element {
  position: relative;

  &__helper {
    font-size: $font-size-small;
  }

  /**
   * Creates label styles for our form elements
   *
   * @selector .slds-form-element__label
   * @restrict .slds-form-element label, .slds-form-element span, .slds-form-element legend, .slds-form-element div, [class*='slds-checkbox'] span, [class*='slds-radio'] span
   */
  &__label {
    display: inline-block;
    color: $color-text-label;
    font-size: $form-label-font-size;
    margin-right: $spacing-x-small;
    padding-top: $spacing-xx-small;
    margin-bottom: $spacing-xxx-small;

    &:empty {
      margin: 0;
    }
  }

  /**
   * Wrapper to any form display element
   *
   * @selector .slds-form-element__control
   * @restrict .slds-form-element div
   */
  &__control {

    // Force radios and checkboxes to be on own line when there
    // are multiples in a group
    .slds-radio,
    .slds-checkbox {
      display: block;
    }
  }

  /**
   * When an icon sits within a form element wrapper and adjacent to another
   * element inside that wrapper such as a .form-element__label
   *
   * @selector .slds-form-element__icon
   * @restrict .slds-form-element div
   */
  &__icon {
    display: inline-block;
    position: relative; // Since this container can have a tooltip
    vertical-align: text-bottom;
  }

  /**
   * Creates inline help styles, sits below .form-element__control
   *
   * @selector .slds-form-element__help
   * @restrict .slds-form-element div
   */
  &__help {
    font-size: $font-size-small;
    margin-top: $spacing-xxx-small;
    display: block;
  }

  // Inline Edit on static form element
  &_edit,
  &--edit {

    .slds-form-element__static {
      width: calc(100% - #{$square-icon-small-boundary}); // Leave space for edit icon
    }

    .slds-button_icon {
      position: absolute;
      right: 0;
      bottom: 0;
    }
  }
}


/**
 * Fixed text that sits to the left or right of an input
 *
 * @selector .slds-form-element__addon
 * @restrict .slds-form-element span
 */
.slds-form-element__addon {
  display: inline-block;
  margin: 0 $spacing-x-small;
  align-self: center;
}

/**
 * Read-only components are used to display immutable data within a form
 *
 * @summary Initializes read-only form element
 *
 * @selector .slds-form-element__static
 * @restrict .slds-form-element span, .slds-form-element div
 */
.slds-form-element__static {
  display: inline-block;
  font-size: $input-static-font-size;
  font-weight: $input-static-font-weight;
  color: $input-static-color;

  &.slds-text-longform *:last-child {
    margin-bottom: 0;
  }

  /**
   * Inline Edit on static form element
   *
   * @selector .slds-form-element__static_edit
   * @restrict .slds-form-element__static
   */
  &--edit {
    width: calc(100% - #{$square-icon-small-boundary}); // Leave space for edit icon
  }
}

/**
 * Required Star
 *
 * @selector .slds-required
 * @restrict .slds-form-element abbr, abbr
 */
.slds-required {
  color: $color-text-error;
  margin: 0 $spacing-xxx-small;
}

/**
 * Error styles for form element
 *
 * @selector .slds-has-error
 * @restrict .slds-form-element
 * @modifier
 * @group feedback
 */
.slds-has-error {

  .slds-form-element__help {
    color: $color-text-error;
  }
}
