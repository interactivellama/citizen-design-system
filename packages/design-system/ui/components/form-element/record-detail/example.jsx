// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import RecordDetail from './';
import * as Snapshot from './snapshots.data';

export default (
  <RecordDetail
    direction="stacked"
    snapshot={Snapshot.ObjectFieldTypes}
    isViewMode
    hasInlineEdit
  />
);

export let examples = [
  {
    id: 'stacked-single-column',
    label: 'Stacked form layout - 1 column - Read Only Mode',
    element: (
      <RecordDetail
        direction="stacked"
        snapshot={Snapshot.ObjectFieldTypesSingleColumn}
        isViewMode
        isSingleColumn
      />
    )
  },
  {
    id: 'edit-stacked',
    label: 'Stacked form layout - Edit Mode',
    element: (
      <RecordDetail direction="stacked" snapshot={Snapshot.ObjectFieldTypes} />
    )
  },
  {
    id: 'horizontal',
    label: 'Horizontal form layout - View Mode',
    element: (
      <RecordDetail
        direction="horizontal"
        snapshot={Snapshot.ObjectFieldTypesHorizontal}
        isViewMode
        hasInlineEdit
      />
    )
  },
  {
    id: 'horizontal-single-column',
    label: 'Horizontal form layout - 1 column - Read Only Mode',
    element: (
      <RecordDetail
        direction="horizontal"
        snapshot={Snapshot.ObjectFieldTypesSingleColumn}
        isViewMode
        isSingleColumn
      />
    )
  },
  {
    id: 'edit-horizontal',
    label: 'Horizontal form layout - Edit Mode',
    element: (
      <RecordDetail
        direction="horizontal"
        snapshot={Snapshot.ObjectFieldTypesHorizontal}
      />
    )
  }
];
