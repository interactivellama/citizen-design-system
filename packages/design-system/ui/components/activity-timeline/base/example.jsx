// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from 'ui/components/button-icons/base/example';
import { ActivityTimelineItem as Event } from 'ui/components/activity-timeline/event/example';
import { ActivityTimelineItem as Call } from 'ui/components/activity-timeline/call/example';
import { ActivityTimelineItem as Email } from 'ui/components/activity-timeline/email/example';
import { ActivityTimelineItem as Task } from 'ui/components/activity-timeline/task/example';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

export let TimelineIcon = props =>
  <div
    className={classNames(
      'slds-icon_container',
      props.symbol == 'log_a_call' ?
        'slds-icon-standard-log-a-call'
      : 'slds-icon-standard-' + props.symbol
    )}
    title={props.symbol == 'log_a_call' ? 'call' : props.symbol }
  >
    <SvgIcon
      className="slds-icon slds-icon--small"
      sprite="standard"
      symbol={ props.symbol }
    />
  </div>;

export let TimelineActions = props =>
  <div className="slds-timeline__actions">
    <p className="slds-timeline__date">{ props.date || 'Feb 24' }</p>
    <ButtonIcon
      className="slds-button--icon-border-filled slds-button--icon-x-small"
      symbol="down"
      assistiveText={ props.assistiveText || 'More Options' }
      aria-haspopup="true"
      title={ props.title || 'More Options' }
    />
  </div>;

export default (
  <ul className="slds-timeline">
    <li>
      <Task />
    </li>
    <li>
      <Event />
    </li>
    <li>
      <Call />
    </li>
    <li>
      <Email />
    </li>
  </ul>
);
