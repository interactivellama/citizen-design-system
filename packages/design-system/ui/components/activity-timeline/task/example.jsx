// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import MediaObject from 'ui/utilities/media-objects/index.react';
import { Checkbox } from 'ui/components/checkbox/base/example';
import { TimelineIcon, TimelineActions } from 'ui/components/activity-timeline/base/example';

export let ActivityTimelineItem = props =>
  <div className="slds-timeline__item">
    <span className="slds-assistive-text">Task</span>
    <MediaObject
      figureRight={
        <TimelineActions
          title="More Options for Task, Review proposals"
          assistiveText="More Options for Task, Review proposals"
        />
      }
    >
      <MediaObject
        className="slds-timeline__media slds-timeline__media--task"
        figureLeft={ <TimelineIcon symbol="task" /> }
        figureLeftClassName="slds-timeline__icon"
      >
        <MediaObject
          title="Review proposals for EBC deck with larger team and have marketing review this"
          figureLeft={ <Checkbox hideLabel label="mark-complete" /> }
        >
          <h3 className="slds-truncate" title="Review proposals for EBC deck with larger team and have marketing review this">
            <a href="javascript:void(0);">Review proposals for EBC deck with larger team and have marketing review this</a>
          </h3>
          <ul className="slds-list--horizontal slds-wrap">
            <li className="slds-m-right--large">
              <span className="slds-text-title">Contact:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Lei Chan</a></span>
            </li>
            <li className="slds-m-right--large">
              <span className="slds-text-title">Assigned to:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Betty Mason</a></span>
            </li>
          </ul>
        </MediaObject>
      </MediaObject>
    </MediaObject>
  </div>;

export default (
  <ActivityTimelineItem />
);
