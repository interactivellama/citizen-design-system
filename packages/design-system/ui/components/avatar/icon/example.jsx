// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from 'app_modules/ui/svg-icon';
import { StandardIcon } from 'ui/components/icons/standard/example';
import { Avatar } from 'ui/components/avatar/base/example';
import classNames from 'classnames';


//////////////////////////////////////////////
// Export
//////////////////////////////////////////////

export default (
  <Avatar>
    <StandardIcon />
  </Avatar>
);

export let states = [
  {
    id: 'avatar-square-icon',
    label: 'Square Icon Fallback',
    element:
      <Avatar className="slds-avatar--medium">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'avatar-circle-icon',
    label: 'Circle Icon Fallback',
    element:
      <Avatar className="slds-avatar--circle slds-avatar--medium">
        <StandardIcon symbol="user" />
      </Avatar>
  },
  {
    id: 'size-x-small',
    label: 'Size - X Small',
    element:
      <Avatar className="slds-avatar--x-small">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-small',
    label: 'Size - Small',
    element:
      <Avatar className="slds-avatar--small">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-medium',
    label: 'Size - Medium',
    element:
      <Avatar className="slds-avatar--medium">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-large',
    label: 'Size - Large',
    element:
      <Avatar className="slds-avatar--large">
        <StandardIcon />
      </Avatar>
  }
];
