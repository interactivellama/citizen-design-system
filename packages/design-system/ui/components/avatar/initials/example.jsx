// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default (
  <Avatar>
    <AvatarInitials
      iconClassName="slds-icon-standard-account"
      title="company name"
      initials="Ac"
    />
  </Avatar>
);

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="javascript:void(0);">
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
