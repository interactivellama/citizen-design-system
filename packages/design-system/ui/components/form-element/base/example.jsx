// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { FormElement, FormElementLabel } from '../';
import { Input } from '../../input/base/example';
import { Textarea } from '../../textarea/base/example';
import { Checkbox } from '../../checkbox/base/example';
import { Radio } from '../../radio-group/base/example';
import { Form } from '../../form-layout/base/example';
import { CompoundForm } from '../../form-layout/compound/example';

const inputLabel = 'Form Element Label';
const defaultInputId = 'input-unique-id';
const errorId = 'error-message-unique-id';

export default (
  <FormElement labelContent={inputLabel} inputId={defaultInputId}>
    <Input id={defaultInputId} />
  </FormElement>
);

export let states = [
  {
    id: 'required',
    label: 'Required',
    element: (
      <FormElement
        labelContent={inputLabel}
        inputId={defaultInputId}
        isRequired
      >
        <Input id={defaultInputId} required />
      </FormElement>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <FormElement
        formElementClassName="slds-has-error"
        labelContent={inputLabel}
        inputId={defaultInputId}
        errorId={errorId}
        isRequired
        inlineMessage="This field is required"
      >
        <Input id={defaultInputId} required aria-describedby={errorId} />
      </FormElement>
    )
  }
];

export let examples = [
  {
    id: 'input',
    label: 'Input',
    element: (
      <FormElement labelContent={inputLabel} inputId={defaultInputId}>
        <Input id={defaultInputId} />
      </FormElement>
    )
  },
  {
    id: 'textarea',
    label: 'Textarea',
    element: (
      <FormElement labelContent={inputLabel} inputId={defaultInputId}>
        <Textarea id={defaultInputId} />
      </FormElement>
    )
  },
  {
    id: 'checkbox',
    label: 'Checkbox',
    element: (
      <FormElement>
        <Checkbox label="Checkbox Label" />
      </FormElement>
    )
  },
  {
    id: 'radio-group',
    label: 'Radio Group',
    element: (
      <fieldset className="slds-form-element">
        <legend className="slds-form-element__legend slds-form-element__label">
          Form Element Label
        </legend>
        <div className="slds-form-element__control">
          <Radio checked label="Radio Label One" />
          <Radio label="Radio Label Two" />
        </div>
      </fieldset>
    )
  },
  {
    id: 'inline-help',
    label: 'Inline Help',
    element: (
      <FormElement
        labelContent={inputLabel}
        inputId={defaultInputId}
        inlineMessage="ex: (415)111-2222"
      >
        <Input id={defaultInputId} />
      </FormElement>
    )
  },
  {
    id: 'tooltip-help',
    label: 'Tooltip Help',
    element: (
      <div
        style={{
          paddingLeft: '2rem',
          paddingTop: '3rem',
          position: 'relative'
        }}
      >
        <FormElement
          labelContent={inputLabel}
          inputId={defaultInputId}
          hasTooltip
        >
          <Input id={defaultInputId} />
        </FormElement>
      </div>
    )
  },
  {
    id: 'form-layout',
    label: 'Form Layout',
    element: <Form isStacked />
  },
  {
    id: 'form-layout-horizontal',
    label: 'Form Layout - Horizontal',
    element: <Form isHorizontal />
  },
  {
    id: 'compound-form',
    label: 'Compound Form',
    element: <CompoundForm />
  }
];
