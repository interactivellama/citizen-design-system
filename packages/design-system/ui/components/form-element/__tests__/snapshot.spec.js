// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import {
  SimpleFormElementWrapper,
  FormElementControl,
  FormElementLabel,
  FormElementSpanFauxLabel,
  FormElementTooltip,
  FormElementHelpMessage,
  FormElementStatic,
  Fieldset,
  FieldsetWrapper,
  Legend,
  FormElement
} from '../';
import { Input } from '../../input/base/example';
import ButtonIcon from '../../button-icons/';

const { matchesMarkup } = createHelpers(__dirname);

describe('Form Element', () => {
  it('renders form element wrapper', () =>
    matchesMarkup(
      <SimpleFormElementWrapper>content</SimpleFormElementWrapper>
    ));
  it('renders form element wrapper with inline edit classname', () =>
    matchesMarkup(
      <SimpleFormElementWrapper isEditable>content</SimpleFormElementWrapper>
    ));
  it('renders form element wrapper with view mode classname', () =>
    matchesMarkup(
      <SimpleFormElementWrapper isViewMode>content</SimpleFormElementWrapper>
    ));
  it('renders form element wrapper with hint classname', () =>
    matchesMarkup(
      <SimpleFormElementWrapper hasHint>content</SimpleFormElementWrapper>
    ));
  it('renders form element wrapper with error classname', () =>
    matchesMarkup(
      <SimpleFormElementWrapper hasError>content</SimpleFormElementWrapper>
    ));
  it('renders form element label', () =>
    matchesMarkup(<FormElementLabel labelContent="Form label" />));
  it('renders form element label with required asterisk', () =>
    matchesMarkup(<FormElementLabel labelContent="Form label" isRequired />));
  it('renders form element label with required asterisk', () =>
    matchesMarkup(<FormElementLabel labelContent="Form label" isRequired />));
  it('renders form element label with hidden label', () =>
    matchesMarkup(
      <FormElementLabel labelContent="Form label" hasHiddenLabel />
    ));
  it('renders form element label with label ID', () =>
    matchesMarkup(
      <FormElementLabel labelContent="Form label" labelId="label-id" />
    ));
  it('renders form element label with label classname', () =>
    matchesMarkup(
      <FormElementLabel labelContent="Form label" labelClassName="slds-foo" />
    ));
  it('renders form element label with input ID', () =>
    matchesMarkup(
      <FormElementLabel labelContent="Form label" inputId="input-id" />
    ));
  it('renders form element label as a span element', () =>
    matchesMarkup(
      <FormElementSpanFauxLabel>content</FormElementSpanFauxLabel>
    ));
  it('renders form element control', () =>
    matchesMarkup(<FormElementControl>content</FormElementControl>));
  it('renders form element control with classname', () =>
    matchesMarkup(
      <FormElementControl className="slds-foo">content</FormElementControl>
    ));
  it('renders form element help text icon', () =>
    matchesMarkup(<FormElementTooltip />));
  it('renders form element help text icon with a tooltip', () =>
    matchesMarkup(<FormElementTooltip showTooltip />));
  it('renders form element inline help message', () =>
    matchesMarkup(<FormElementHelpMessage />));
  it('renders form element inline help message with ID', () =>
    matchesMarkup(<FormElementHelpMessage errorId="error-id" />));
  it('renders form element with static/view only value', () =>
    matchesMarkup(<FormElementStatic>content</FormElementStatic>));
  it('renders form element with static/view only longform value', () =>
    matchesMarkup(<FormElementStatic isLongform>content</FormElementStatic>));
  it('renders form element fieldset wrapper', () =>
    matchesMarkup(<FieldsetWrapper>content</FieldsetWrapper>));
  it('renders form element fieldset wrapper with compound fields', () =>
    matchesMarkup(
      <FieldsetWrapper hasCompoundFields>content</FieldsetWrapper>
    ));
  it('renders form element fieldset wrapper with error classname', () =>
    matchesMarkup(<FieldsetWrapper hasError>content</FieldsetWrapper>));
  it('renders form element fieldset wrapper with required classname', () =>
    matchesMarkup(<FieldsetWrapper isRequired>content</FieldsetWrapper>));
  it('renders form element fieldset legend', () =>
    matchesMarkup(<Legend>content</Legend>));
  it('renders form element fieldset legend with required asterisk', () =>
    matchesMarkup(<Legend isRequired>content</Legend>));
  it('renders form element fieldset legend with help text icon', () =>
    matchesMarkup(<Legend hasTooltip>content</Legend>));
  it('renders form element fieldset legend that is visually hidden', () =>
    matchesMarkup(<Legend hasHiddenLabel>content</Legend>));
  it('renders form element as a fieldset', () =>
    matchesMarkup(<Fieldset label="Fieldset legend">content</Fieldset>));
  it('renders form element as a fieldset with required asterisk', () =>
    matchesMarkup(
      <Fieldset label="Fieldset legend" isRequired>
        content
      </Fieldset>
    ));
  it('renders form element as a fieldset with error classname', () =>
    matchesMarkup(
      <Fieldset label="Fieldset legend" hasError>
        content
      </Fieldset>
    ));
  it('renders form element as a fieldset with compound field classname', () =>
    matchesMarkup(
      <Fieldset label="Fieldset legend" hasCompoundFields>
        content
      </Fieldset>
    ));
  it('renders form element as a fieldset with help text icon', () =>
    matchesMarkup(
      <Fieldset label="Fieldset legend" hasTooltip>
        content
      </Fieldset>
    ));
  it('renders form element as a fieldset with hidden legend', () =>
    matchesMarkup(
      <Fieldset label="Fieldset legend" hasHiddenLabel>
        content
      </Fieldset>
    ));
});

describe('Form Element Composition', () => {
  it('renders form element with input', () =>
    matchesMarkup(
      <FormElement labelContent="Form label" inputId="input-1">
        <Input id="input-1" />
      </FormElement>
    ));
  it('renders form element with input that is required', () =>
    matchesMarkup(
      <FormElement labelContent="Form label" inputId="input-1" isRequired>
        <Input id="input-1" required />
      </FormElement>
    ));
  it('renders form element with input that has an error', () =>
    matchesMarkup(
      <FormElement
        labelContent="Form label"
        inputId="input-1"
        errorId="error-1"
        isRequired
        hasError
        inlineMessage="This field is required"
      >
        <Input id="input-1" required aria-describedby="error-1" />
      </FormElement>
    ));
  it('renders inline editable form element in view mode', () =>
    matchesMarkup(
      <FormElement isViewMode labelContent="Status" isEditable hasHint>
        <FormElementStatic>In Progress</FormElementStatic>
        <ButtonIcon
          iconClassName="slds-button__icon_hint"
          symbol="edit"
          assistiveText="Edit: Status"
          title="Edit: Status"
        />
      </FormElement>
    ));
});
