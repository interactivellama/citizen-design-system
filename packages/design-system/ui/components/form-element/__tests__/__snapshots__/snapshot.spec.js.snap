// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Element Composition renders form element with input 1`] = `
<div
  className="slds-form-element"
  role={undefined}
>
  <label
    className="slds-form-element__label"
    htmlFor="input-1"
    id={undefined}
  >
    Form label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      defaultValue={undefined}
      id="input-1"
      placeholder={undefined}
      readOnly={undefined}
      type="text"
    />
  </div>
</div>
`;

exports[`Form Element Composition renders form element with input that has an error 1`] = `
<div
  className="slds-form-element slds-has-error"
  role={undefined}
>
  <label
    className="slds-form-element__label"
    htmlFor="input-1"
    id={undefined}
  >
    <abbr
      className="slds-required"
      title="required"
    >
      * 
       
    </abbr>
    Form label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      aria-describedby="error-1"
      className="slds-input"
      defaultValue={undefined}
      id="input-1"
      placeholder={undefined}
      readOnly={undefined}
      required={true}
      type="text"
    />
  </div>
  <div
    className="slds-form-element__help"
    id="error-1"
  >
    This field is required
  </div>
</div>
`;

exports[`Form Element Composition renders form element with input that is required 1`] = `
<div
  className="slds-form-element"
  role={undefined}
>
  <label
    className="slds-form-element__label"
    htmlFor="input-1"
    id={undefined}
  >
    <abbr
      className="slds-required"
      title="required"
    >
      * 
       
    </abbr>
    Form label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      defaultValue={undefined}
      id="input-1"
      placeholder={undefined}
      readOnly={undefined}
      required={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Form Element Composition renders inline editable form element in view mode 1`] = `
<div
  className="slds-form-element slds-form-element_edit slds-form-element_readonly slds-hint-parent"
  role={undefined}
>
  <span
    className="slds-form-element__label"
    id={undefined}
  >
    Status
  </span>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-form-element__static"
    >
      In Progress
    </div>
    <button
      aria-haspopup={undefined}
      className="slds-button slds-button_icon"
      id={undefined}
      title="Edit: Status"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-button__icon_hint"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#edit"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Edit: Status
      </span>
    </button>
  </div>
</div>
`;

exports[`Form Element renders form element as a fieldset 1`] = `
<fieldset
  className="slds-form-element"
  id={undefined}
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Fieldset legend
  </legend>
  <div
    className="slds-form-element__control"
  >
    content
  </div>
</fieldset>
`;

exports[`Form Element renders form element as a fieldset with compound field classname 1`] = `
<fieldset
  className="slds-form-element slds-form_compound"
  id={undefined}
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Fieldset legend
  </legend>
  <div
    className="slds-form-element__control"
  >
    content
  </div>
</fieldset>
`;

exports[`Form Element renders form element as a fieldset with error classname 1`] = `
<fieldset
  className="slds-form-element slds-has-error"
  id={undefined}
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Fieldset legend
  </legend>
  <div
    className="slds-form-element__control"
  >
    content
  </div>
</fieldset>
`;

exports[`Form Element renders form element as a fieldset with help text icon 1`] = `
<fieldset
  className="slds-form-element"
  id={undefined}
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Fieldset legend
  </legend>
  <div
    className="slds-form-element__icon"
  >
    <button
      aria-haspopup={undefined}
      className="slds-button slds-button_icon"
      id={undefined}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Help
      </span>
    </button>
  </div>
  <div
    className="slds-form-element__control"
  >
    content
  </div>
</fieldset>
`;

exports[`Form Element renders form element as a fieldset with hidden legend 1`] = `
<fieldset
  className="slds-form-element"
  id={undefined}
>
  <legend
    className="slds-form-element__legend slds-form-element__label slds-assistive-text"
  >
    Fieldset legend
  </legend>
  <div
    className="slds-form-element__control"
  >
    content
  </div>
</fieldset>
`;

exports[`Form Element renders form element as a fieldset with required asterisk 1`] = `
<fieldset
  className="slds-form-element slds-is-required"
  id={undefined}
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    <abbr
      className="slds-required"
      title="required"
    >
      * 
       
    </abbr>
    Fieldset legend
  </legend>
  <div
    className="slds-form-element__control"
  >
    content
  </div>
</fieldset>
`;

exports[`Form Element renders form element control 1`] = `
<div
  className="slds-form-element__control"
>
  content
</div>
`;

exports[`Form Element renders form element control with classname 1`] = `
<div
  className="slds-form-element__control slds-foo"
>
  content
</div>
`;

exports[`Form Element renders form element fieldset legend 1`] = `
<legend
  className="slds-form-element__legend slds-form-element__label"
>
  content
</legend>
`;

exports[`Form Element renders form element fieldset legend that is visually hidden 1`] = `
<legend
  className="slds-form-element__legend slds-form-element__label slds-assistive-text"
>
  content
</legend>
`;

exports[`Form Element renders form element fieldset legend with help text icon 1`] = `
Array [
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    content
  </legend>,
  <div
    className="slds-form-element__icon"
  >
    <button
      aria-haspopup={undefined}
      className="slds-button slds-button_icon"
      id={undefined}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Help
      </span>
    </button>
  </div>,
]
`;

exports[`Form Element renders form element fieldset legend with required asterisk 1`] = `
<legend
  className="slds-form-element__legend slds-form-element__label"
>
  <abbr
    className="slds-required"
    title="required"
  >
    * 
     
  </abbr>
  content
</legend>
`;

exports[`Form Element renders form element fieldset wrapper 1`] = `
<fieldset
  className="slds-form-element"
  id={undefined}
>
  content
</fieldset>
`;

exports[`Form Element renders form element fieldset wrapper with compound fields 1`] = `
<fieldset
  className="slds-form-element slds-form_compound"
  id={undefined}
>
  content
</fieldset>
`;

exports[`Form Element renders form element fieldset wrapper with error classname 1`] = `
<fieldset
  className="slds-form-element slds-has-error"
  id={undefined}
>
  content
</fieldset>
`;

exports[`Form Element renders form element fieldset wrapper with required classname 1`] = `
<fieldset
  className="slds-form-element slds-is-required"
  id={undefined}
>
  content
</fieldset>
`;

exports[`Form Element renders form element help text icon 1`] = `
<div
  className="slds-form-element__icon"
>
  <button
    aria-describedby={undefined}
    aria-haspopup={undefined}
    className="slds-button slds-button_icon"
    id={undefined}
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Help
    </span>
  </button>
</div>
`;

exports[`Form Element renders form element help text icon with a tooltip 1`] = `
<div
  className="slds-form-element__icon"
>
  <button
    aria-describedby="help"
    aria-haspopup={undefined}
    className="slds-button slds-button_icon"
    id={undefined}
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Help
    </span>
  </button>
  <div
    className="slds-popover slds-popover_tooltip slds-nubbin_bottom-left"
    id="help"
    role="tooltip"
    style={
      Object {
        "left": "-15px",
        "position": "absolute",
        "top": "-45px",
        "width": "170px",
      }
    }
  >
    <div
      className="slds-popover__body"
    >
      Some helpful information
    </div>
  </div>
</div>
`;

exports[`Form Element renders form element inline help message 1`] = `
<div
  className="slds-form-element__help"
  id={undefined}
/>
`;

exports[`Form Element renders form element inline help message with ID 1`] = `
<div
  className="slds-form-element__help"
  id="error-id"
/>
`;

exports[`Form Element renders form element label 1`] = `
<label
  className="slds-form-element__label"
  htmlFor={undefined}
  id={undefined}
>
  Form label
</label>
`;

exports[`Form Element renders form element label as a span element 1`] = `
<span
  className="slds-form-element__label"
  id={undefined}
>
  content
</span>
`;

exports[`Form Element renders form element label with hidden label 1`] = `
<label
  className="slds-form-element__label slds-assistive-text"
  htmlFor={undefined}
  id={undefined}
>
  Form label
</label>
`;

exports[`Form Element renders form element label with input ID 1`] = `
<label
  className="slds-form-element__label"
  htmlFor="input-id"
  id={undefined}
>
  Form label
</label>
`;

exports[`Form Element renders form element label with label ID 1`] = `
<label
  className="slds-form-element__label"
  htmlFor={undefined}
  id="label-id"
>
  Form label
</label>
`;

exports[`Form Element renders form element label with label classname 1`] = `
<label
  className="slds-form-element__label slds-foo"
  htmlFor={undefined}
  id={undefined}
>
  Form label
</label>
`;

exports[`Form Element renders form element label with required asterisk 1`] = `
<label
  className="slds-form-element__label"
  htmlFor={undefined}
  id={undefined}
>
  <abbr
    className="slds-required"
    title="required"
  >
    * 
     
  </abbr>
  Form label
</label>
`;

exports[`Form Element renders form element label with required asterisk 2`] = `
<label
  className="slds-form-element__label"
  htmlFor={undefined}
  id={undefined}
>
  <abbr
    className="slds-required"
    title="required"
  >
    * 
     
  </abbr>
  Form label
</label>
`;

exports[`Form Element renders form element with static/view only longform value 1`] = `
<div
  className="slds-form-element__static slds-text-longform"
>
  content
</div>
`;

exports[`Form Element renders form element with static/view only value 1`] = `
<div
  className="slds-form-element__static"
>
  content
</div>
`;

exports[`Form Element renders form element wrapper 1`] = `
<div
  className="slds-form-element"
  role={undefined}
>
  content
</div>
`;

exports[`Form Element renders form element wrapper with error classname 1`] = `
<div
  className="slds-form-element slds-has-error"
  role={undefined}
>
  content
</div>
`;

exports[`Form Element renders form element wrapper with hint classname 1`] = `
<div
  className="slds-form-element slds-hint-parent"
  role={undefined}
>
  content
</div>
`;

exports[`Form Element renders form element wrapper with inline edit classname 1`] = `
<div
  className="slds-form-element slds-form-element_edit"
  role={undefined}
>
  content
</div>
`;

exports[`Form Element renders form element wrapper with view mode classname 1`] = `
<div
  className="slds-form-element slds-form-element_readonly"
  role={undefined}
>
  content
</div>
`;
